### DISCUSSION TITLE
PrimeFX with PrimeOne for FXML (JavaFX/GluonFX)

### DISCUSSION THREAD

**Discussion by rrangelo** [primefaces/community#388] [PrimeFaces] [answered]
Hi! Have you ever think create PrimeFX with PrimeOne for FXML (JavaFX/GluonFX)?

**Comment by melloware** [comment]
@rrangelo I am not familiar with JavaFX or what it would take to implement do you have articles or anything that can be done.  

@tmulle you are familiar with JavaFX and Gluon is this even possible?

**Reply by tmulle** [reply, marked as answer]
I'm not an expert, but I think it might be technically possible.
Although, JavaFX doesn't use true CSS naming for their variables in CSS so I think some conversion would need to happen.
It wouldn't be a 1-1 conversion.

As far as Gluon, I believe you are locked into their styling (material theme) when you use their components in the Mobile product.
But if you are not using their mobile components then I would think you could use your own styling.

Just speculation, I haven't tried any of it.

This is a link to the JavaFX CSS guide: https://docs.oracle.com/javafx/2/api/javafx/scene/doc-files/cssref.html

Also, here's a good website for JavaFX related items.
https://www.jfx-central.com

**Reply by rrangelo** [reply]
Hi all!

@melloware, off course!

- [OpenJFX](https://openjfx.io/)
- [Gluon](https://gluonhq.com/)

Here an example about UI controls implementation, it can be taken like guide:
- [ControlsFX](https://controlsfx.github.io/)

@tmulle, now, in new implementations, JavaFX support CSS, here an example:

[Gluon Sample: notes](https://github.com/gluonhq/gluon-samples/tree/master/notes)

In that repository there are all the models to implement a project with GluonFX

Any other questions, I'm at your service.

**Reply by melloware** [reply]
From looking at these components it would have to be a complete rewrite of PrimeFaces into these types of controls.  There is no simple shim that could be applied.

**Reply by rrangelo** [reply]
Oh! I understand, I was thinking of something simpler that could enable the use of PrimeOne Design in JavaFX applications, like [FluentFX](https://github.com/Ivan-Kalatchev/FluentFX) with Fluent Design over ControlsFX, or like [MaterialFX](https://github.com/palexdev/MaterialFX) with Material Design.
